Refactored Float and Integer field types to use external validators

alongosz opened the pull request:
| :ticket: Issue | Related to IBX-8138 |
|----------------|-----------|

#### Related PRs: 
- https://github.com/ibexa/core/pull/385

#### Description:

This PR delegates redundant validation code in Float and Integer field types to proper, already existing validators.
Also refactored tests, including tests for String validator due to similarities.

I've moved scope-affected Integer, Float, and String Validator test classes to the correct Validator namespace, reflecting better src namespace structure.

#### For QA:

Regression build should be enough (https://github.com/ibexa/commerce/pull/1010).

adamwojs commented on tests/lib/FieldType/Validator/BaseNumericValidatorTestCase.php:
```diff
@@ -0,0 +1,248 @@
+                [
+                    $maxNumericValueName => new \DateTime(),
+                    $minNumericValueName => 1234,
+                ],
+                [$maxValueNumericTypeValidationMessage],
+            ],
+            [
+                [
+                    $minNumericValueName => true,
+                    $maxNumericValueName => 1234,
+                ],
+                [$minValueNumericTypeValidationMessage],
+                [
+                    ['%parameter%' => $minNumericValueName],
+                ],
+            ],
+            [
+                [
+                    $minNumericValueName => self::WRONG_NUMERIC_MIN_VALUE,
+                    $maxNumericValueName => self::WRONG_NUMERIC_MAX_VALUE,
+                ],
+                [
+                    $minValueNumericTypeValidationMessage,
+                    $maxValueNumericTypeValidationMessage,
+                ],
+            ],
+            [
+                [
+                    'brljix' => 12345,
+                ],
+                [self::UNKNOWN_PARAM_VALIDATION_MESSAGE],
+            ],
+            [
+                [
+                    $minNumericValueName => 12345,
+                    'brljix' => 12345,
+                ],
+                [self::UNKNOWN_PARAM_VALIDATION_MESSAGE],
+            ],
+        ];
```
Can we provide meaningful names for cases ?

alongosz replied:
> Can we provide meaningful names for cases ?

Fixed via [9a56748](https://github.com/ibexa/core/pull/425/commits/9a5674807311d38a66422fae7b78a429a18819f5).

adamwojs commented on tests/lib/FieldType/Validator/FloatValueValidatorTest.php:
```diff
@@ -0,0 +1,162 @@
+
+    protected function getMinFloatValue(): float
+    {
+        return 10 / 7;
```
Why you decided to use `10 / 7` syntax instead of 1.4285... ?

alongosz replied:
> Why you decided to use `10 / 7` syntax instead of 1.4285... ?

This is just moved code. Probably the author decided it's more readable. I don't see a reason not to use literal computed value. I've changed all occurrences of division into literal values via [9a56748](https://github.com/ibexa/core/pull/425/commits/9a5674807311d38a66422fae7b78a429a18819f5) :+1:

Steveb-p replied:
It's test code, so readability should go above test performance (negligible too).

But otherwise I'm okay with both variants.

Steveb-p commented on src/lib/FieldType/Validator/IntegerValueValidator.php:
```diff
@@ -9,16 +9,16 @@
 use Ibexa\Core\FieldType\ValidationError;
-use Ibexa\Core\FieldType\Validator;
 use Ibexa\Core\FieldType\Value as BaseValue;
+use function is_int;
```
You didn't import `is_numeric` in the other part of this PR. Why import this one? By mistake?

alongosz replied:
Uhm, not sure what happened there, rolled back via [a887e22](https://github.com/ibexa/core/pull/425/commits/a887e22f53ee954528e2b6315fec9409f96e1a66).

alongosz commented:
Update: added via [a887e22](https://github.com/ibexa/core/pull/425/commits/a887e22f53ee954528e2b6315fec9409f96e1a66) missing return types to `FloatValueValidator::validate` and `IntegerValueValidator::validate` methods and simplified PHPDocs.
